#!/usr/bin/env python

PACKAGE = "rviz_obstacle_generator"  # Replace with your package name

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("speed_km_p_h", double_t, 0, "speed of the dynamic obstacle", 40.0, 0.1, 120.0)
gen.add("angular_velocity_deg_p_sec", double_t, 0, "angular velocity of the dynamic obstacle", 0.0, -30.0, 30.0)

gen.add("duration_sec_static", double_t, 0, "duration time of the static obstacle", 3.0, 0.0, 20.0)
gen.add("duration_sec_dynamic", double_t, 0, "duration time of the dynamic obstacle", 3.0, 0.0, 20.0)
gen.add("x_size_meter_static", double_t, 0, "X size of the static obstacle", 1.0, 0.0, 10.0)
gen.add("y_size_meter_static", double_t, 0, "Y size of the static obstacle", 1.0, 0.0, 10.0)
gen.add("x_size_meter_dynamic", double_t, 0, "X size of the dynamic obstacle", 4.650, 0.0, 10.0)
gen.add("y_size_meter_dynamic", double_t, 0, "Y size of the dynamic obstacle", 1.825, 0.0, 10.0)

exit(gen.generate(PACKAGE, "Obstacle", "Obstacle"))
